### PR TITLE
fix(masks): time-picker, date-picker, date-range-picker > updates to mask behavior

### PR DESCRIPTION
**Pull request by jake-crawford-tyler**
## PR Checklist

Please check if your PR fulfills the following requirements:

- Tests for the changes have been added/updated: [Y]
- Docs have been added/updated: [Y]
- Does this PR introduce a breaking change? [N]
- I have linked any related GitHub issues to be closed when this PR is merged? [Y]

## Describe the new behavior?
time-picker, date-picker, and date-range-picker will now only display masks when they are focused. Also masks will hide on blur when no value is populated.

## Additional information
Closes: https://github.com/tyler-technologies-oss/forge/issues/314

### CODE REVIEW

**Review by DRiFTy17 — COMMENTED** [review]
This looks good to me from a functional perspective. It gets me wondering though, do we want to just always leave the mask uninitialized until the field is first focused instead? The input masking functionality is only needed while the user is typing into the field. I like that we no longer have the mask overhead while the field is resting since they are not needed at that time...

A couple of minor issues relating to the floating label jumped out at me as well:
- The label doesn't animate to its floating position when the mask is enabled on focus
- The label jumps slightly when tabbing between the inputs in the date-range-picker

File: src/lib/time-picker/time-picker-foundation.ts [hunk 1]
```diff
@@ -769,9 +772,11 @@ export class TimePickerFoundation implements ITimePickerFoundation {
     if (this._allowSeconds !== value) {
       this._allowSeconds = !!value;
       this._applyAllowSeconds();
-      if (this._isInitialized) {
+      if (this._isInitialized && !this._showMaskFormat) {
```
Comment (madhater) [inline src/lib/time-picker/time-picker-foundation.ts:775]: this seems backwards, if not showMaskFormat then apply the mask?

File: src/test/spec/date-range-picker/date-range-picker.spec.ts [hunk 2]
```diff
@@ -512,6 +512,7 @@ describe('DateRangePickerComponent', function(this: ITestContext) {
       const date = '05/04/2020';
       const expectedDate = new Date(date);
       const inputElement = getToElement(this.context.component);
+      inputElement.focus();
```
Comment (jake-crawford-tyler) [inline src/test/spec/date-range-picker/date-range-picker.spec.ts:515]: Honestly, this one was a bit confusing as the "from" unit test did not require a focus. I am open to suggestions on if this is an incorrect test.

File: src/lib/time-picker/time-picker-foundation.ts (hunk 1 above)
Comment (jake-crawford-tyler) [reply src/lib/time-picker/time-picker-foundation.ts:772]: After re-evaluating. I still think this is correct. If this condition was not here, then applyMask function will make the mask show if showMaskFormat is true. When testing in the UI, turning on and off allowSeconds and use24HourTime will cause the mask to show when it should not. Still open to discussion on this one @madhater and @DRiFTy17

**Comment by jake-crawford-tyler** [comment]
I will need assistance tracking down the issue with the animations. How should I proceed from here? @DRiFTy17

**Comment by DRiFTy17** [comment]
> I will need assistance tracking down the issue with the animations. How should I proceed from here? @DRiFTy17

@jake-crawford-tyler I started looking into this earlier today, but I've been hopping back and forth between a number of different tasks all day... I was able to fix the issue regarding the floating label animation when focusing the field with the mask format visible though.

I still need to track down the issue in the date-range-picker when tabbing between the inputs, and then finish reviewing the latest changes.

Sit tight for a bit and I'll update your PR with some changes, and give it all another review.

**Comment by jake-crawford-tyler** [comment]
Perfect, thank you @DRiFTy17 , I appreciate the help!

**Review by DRiFTy17 — APPROVED** [review]

**Pull request merged**